### PR TITLE
620 Product Resource Tab

### LINKED ISSUE #620: [Product] Product/Product Family detail page - part 4 (coveo)
- [ ] Resource Tab
- [ ] Products Tab (bundle only)
- [ ] Related Products Tab

All using Coevo Atomic library

### PR DESCRIPTION
**Pull request by rgravitvl**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #620 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/us/en/products/family/allegra-v-15r-benchtop-centrifuges
- After: https://620-product-detail-res-tab--danaher-ls-aem--hlxsites.hlx.page/us/en/products/family/allegra-v-15r-benchtop-centrifuges
- After: https://620-product-detail-res-tab--danaher-ls-aem--hlxsites.hlx.live/us/en/products/family/allegra-v-15r-benchtop-centrifuges

### CODE REVIEW

**Review by davenichols-DHLS — APPROVED** [review]
Nice, I can see the data in there for the Resources.  Where we going to defer loading this on click, or is this from the initial Coveo call anyway?

**Comment by rgravitvl** [comment]
> Nice, I can see the data in there for the Resources. Where we going to defer loading this on click, or is this from the initial Coveo call anyway?

Yes @davenichols-DHLS , we will defer loading on click of tab. I guess, same has been already implemented in MolDev and @duynguyen  is gonna work on that. Thanks.

File: templates/productDetail/productDetail.js [hunk 1]
```diff
@@ -17,82 +15,15 @@ function getCoveoApiPayload(qParam) {
...
-  return div({ class: 'vertical-gallery-container' }, div(slideContent, verticalSlides));
-}
-
 export default async function buildAutoBlocks() {
-  let response;
-  if (localStorage.getItem('product-hero')) response = JSON.parse(localStorage.getItem('product-hero'));
-  else {
+  const sku = getMetadata('sku');
+  let response = localStorage.getItem('product-details');
+  if (response && JSON.parse(response).at(0).raw.sku === sku) {
```
Comment (duynguyen) [inline templates/productDetail/productDetail.js:21]: nice to have this check to fix the overly cached response. should we add some null checks to this condition?
e.g. `JSON.parse(response)?.at(0)?.raw?.sku` and maybe a try - catch for the parse.

File: blocks/product-hero/product-hero.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  });
+  const showMore = div({ class: 'view-more' }, 'View More ->');
+  showMore.addEventListener('click', loadMore);
+  verticalSlides.append(showMore);
+  return div({ class: 'vertical-gallery-container' }, div(slideContent, verticalSlides));
+}
+
+export default async function decorate(block) {
+  let response;
+  if (localStorage.getItem('product-details')) response = JSON.parse(localStorage.getItem('product-details'));
```
Comment (duynguyen) [inline blocks/product-hero/product-hero.js:56]: looks like this is used in most product-* blocks, should it go to scripts.js?

File: blocks/product-resources/product-resources.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            </atomic-layout-section>
+        </atomic-search-layout>
+    </atomic-search-interface>
+`;
+
+export default async function decorate(block) {
+  const sku = getMetadata('sku');
+  const host = (window.location.host === 'lifesciences.danaher.com') ? window.location.host : 'stage.lifesciences.danaher.com';
+  let response;
+  if (localStorage.getItem('product-details')) response = JSON.parse(localStorage.getItem('product-details'));
```
Comment (duynguyen) [inline blocks/product-resources/product-resources.js:162]: same comment as above.
looks like this is used in most product-* blocks, should it go to scripts.js?

File: templates/productDetail/productDetail.js (hunk 1 above)
Comment (rgravitvl) [reply templates/productDetail/productDetail.js]: Done

File: blocks/product-resources/product-resources.js (hunk 3 above)
Comment (rgravitvl) [reply blocks/product-resources/product-resources.js]: Done

**Review by duynguyen — APPROVED** [review]

**Pull request merged**